### ISSUE TITLE
Security_Authenticate call in SoapHeader 4 does a double authentication

### ISSUE THREAD

**Issue by arosha445** [amabnl/amadeus-ws-client#151]
When ever i do a **securityAuthenticate** request i get this ERR as a result even after a **securitySignOut**
However after the **securityAuthenticate** request i can do all other requests such as FareMasterPricerCalendar, FareMasterPricerTravelboardSearch etc.

```
"status": "ERR",

"messages": [
{
"code": "16001",
"text": "You have already signed-in. Please sign-out first.",
"level": null
}
],
"response": {
"errorSection": {
"applicationError": {
"errorDetails": {
"errorCode": "16001",
"errorCategory": "EC",
"errorCodeOwner": "LSS"
}
},
"interactiveFreeText": {
"freeTextQualif": {
"subject": "3",
"infoType": "TXT",
"language": "EN"
},
"freeText": "You have already signed-in. Please sign-out first."
}
}
},`

This is what i get after **securitySignOut**

`"status": "OK",
"messages": [],
"response": {
"processStatus": {
"statusCode": "P"
}
},
```

**Comment by DerMika** [comment]
Which SoapHeader is being used in your WSAP?

**Comment by arosha445** [comment]
It's using Soap Header Version  4.0

**Comment by DerMika** [comment]
Ok, 2 things: 

1. Can you provide me with the full SOAP request and responses? See [Logging request and response](https://github.com/amabnl/amadeus-ws-client/blob/master/docs/how-to.rst#logging-request-and-response) in the docs. 
2. It's actually not necessary to send a `Security_Authenticate` message with Soap Header 4. You can just send the message you wish to send, and the library will automatically send authentication data to initialize a new session.

**Comment by DerMika** [comment]
I suspect that the security_authenticate it sends on SoapHeader 4 has auth headers in the headers, as well as authentication data in the message body. That would explain this error.

**Comment by arosha445** [comment]
This is the request and response i got from the log
[Security_AuthenticateRequest](https://gist.github.com/arosha445/36e6e2674600019cdebe91a1f908d725)
[Security_AuthenticateResponse.xml](https://gist.github.com/arosha445/0ea61c80f94a09b42ed7e8d41c4b66b2)

If it's not necessary to send a **Security_Authencticate**, is it also not necessary to do a **securitySignOut** ? Does the library do signout as well?

**Comment by DerMika** [comment]
No, a Signout has to be implemented by you yourself. Otherwise you will have problems when trying to get the application certified by Amadeus (which is needed for production access). That is because an active session uses server resources on Amadeus' infrastructure. 

[Session signout is explained here](https://github.com/amabnl/amadeus-ws-client/blob/master/docs/how-to.rst#ending-a-stateful-session-soap-header-4)

**Comment by DerMika** [comment]
Yes, looking at your request XML, it's as I feared. The library adds authentication headers to the Security_Authenticate message, which in fact makes the message a double authentication message.

I will have to fix that. 

But in the mean time just call the message you need, and let the library handle the authentication.

**Comment by arosha445** [comment]
Yes. I can do all the operations without this. But i was trying to implement a signIn test for phpUnit.  
Anyway thanks a lot @DerMika for your time and consideration.

**Comment by irfanbaigse** [comment]
@DerMika can you pls fix authentication headers ?

**Comment by DerMika** [comment]
This isn't urgent at all, since it's not necessary to do a separate authentication call in SoapHeader 4.

Just call the message you need, and the session will be started by the client.

**Comment by DerMika** [comment]
This bug should be fixed now on master since merging #234. It'll be included in the 1.8.0 release

**Comment by ketan0777** [comment]
Hi

I'm sending soap request getting below error can you guide me for that ? I'm almost new 

<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
    <soap:Body>
        <soap:Fault>
            <faultcode>soap:Client</faultcode>
            <faultstring> 12|Presentation|soap message header incorrect</faultstring>
            <faultactor>SI:muxDZ1</faultactor>
        </soap:Fault>
    </soap:Body>
</soap:Envelope>
